components/Comments/CommentForm.vue

```diff
@@ -76,22 +87,38 @@
       }
     },
     methods: {
+      cancel (form) {
+        if (this.isCommentFormOfContribution) {
+          form.content = ''
+        } else {
+          if (this.replyComment)
+            this.$parent.closeCommentForm()
+        }
+      },
       reply (comment) {
         if (!comment) {
           return
         }
-        try {
-          this.$refs.editor.$refs.editorMentions.insertMention(0, comment.user)
-          this.$scrollTo(this.$refs.editor.$el, 500)
-        } catch (err) {}
+        const comments = this.$parent.$el.classList.contains('comments') ? this.$parent : this.$parent.$parent.$parent
+        if (!this.isExecuted && !comments.hasSubmitted && this.$el.children[1].parentElement.id != 'comment-form') {
+          this.isExecuted = true;
+          this.$nextTick(function () {
+            this.$refs.editor.$refs.editorMentions.insertMention(0, comment.user)
+            this.$scrollTo(this.$el.children[1], 500)
+            setTimeout(() => { this.isExecuted = false; }, 700)
+          })
+        }
       },
       async submitComment () {
+        const comments = this.$parent.$el.classList.contains('comments') ? this.$parent : this.$parent.$parent.$parent
+        comments.hasSubmitted = true
         if (!this.hasContent) {
           this.form.content = ''
           return
         }
         this.isLoading = true
         this.form.contributionId = this.post._id
+        this.form.parentCommentId = this.replyComment ? this.replyComment._id : ''
```

Comment:
Do not use strings, they are not very good when comparing Boolean. Use null for example.

Reply:
Is changed